lib/components/primitive-components/Trace/Trace.ts

```diff
@@ -892,5 +939,60 @@ export class Trace
     })
 
     this.schematic_trace_id = trace.schematic_trace_id
+
+    // Now that the trace is created, check if it should be converted to net labels
+    const complexityInfo = this._getChipSideComplexityInfo()
+    if (complexityInfo !== null) {
+      // Get the pin label and chip reference
+      const chipPort = complexityInfo.chipSidePort
+      const chipComponent = chipPort.parent
+      const chipRefDes = chipComponent?.props.name
+      const pinName = chipPort.props.name || `pin${chipPort.props.pinNumber}`
+      const fullLabel = chipRefDes ? `${chipRefDes}_${pinName}` : pinName
+
+      if (this._countComplexElements() >= 4) {
+        // Create net labels with different formats for chip side vs passive side
+        const chipSidePosition = {
+          position:
+            complexityInfo.chipSidePort._getGlobalSchematicPositionAfterLayout(),
+          facingDirection: complexityInfo.chipSidePort.facingDirection,
+          port: complexityInfo.chipSidePort,
+        }
+
+        const passiveSidePosition = {
+          position:
+            complexityInfo.passiveSidePort._getGlobalSchematicPositionAfterLayout(),
+          facingDirection: complexityInfo.passiveSidePort.facingDirection,
+          port: complexityInfo.passiveSidePort,
+        }
+
+        // Chip side gets just the pin name
+        db.schematic_net_label.insert({
+          text: pinName,
+          source_net_id: chipSidePosition.port.source_port_id!,
+          anchor_position: chipSidePosition.position,
+          center: chipSidePosition.position,
+          anchor_side:
+            getEnteringEdgeFromDirection(chipSidePosition.facingDirection!) ??
+            "bottom",
+        })
+
+        // Passive side gets the full reference (e.g. "U1_OUT1")
+        db.schematic_net_label.insert({
+          text: fullLabel,
+          source_net_id: passiveSidePosition.port.source_port_id!,
+          anchor_position: passiveSidePosition.position,
+          center: passiveSidePosition.position,
+          anchor_side:
+            getEnteringEdgeFromDirection(
+              passiveSidePosition.facingDirection!,
+            ) ?? "bottom",
+        })
```

Comment:
Use the existing function for creating the net labels, just modify it to derive the net label name if it doesnt have schDisplayLabel

Dont allow the schematic trace to be inserted if its complex

Moving the counting function outside this class, its not stateful, you give it the path and it returns the complexElementCount

Reply:
modify the `_doInitialSchematicTraceRenderWithDisplayLabel` method and add the logic to derive the netlabel name if it doesn't have `schDisplayLabel`
<img width="783" alt="Screenshot 2024-12-08 at 9 03 45 AM" src="https://github.com/user-attachments/assets/67491f3a-3ff7-4a10-aafa-8c0f28bbfbbc">